storage/database/db_manager.go

```diff
@@ -55,7 +55,7 @@ type DBManager interface {
 	GetDBConfig() *DBConfig
 	getDatabase(DBEntryType) Database
 	CreateMigrationDBAndSetStatus(blockNum uint64) error
-	FinishStateMigration(succeed bool)
+	FinishStateMigration(succeed bool, endCheck chan struct{})
```

Comment:
Just a minor comment, but in my opinion it would be better if `FinishStateMigration` returns `endCheck` channel, instead of passing it as a parameter.

```
FinishStateMigration(succeed bool) chan struct{}
```

Reply:
Thank you for the comment. I applied your suggestion.